src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -0,0 +1,446 @@
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Green",
+	})
+	public get supportsGreen(): boolean {
+		return this._supportsGreen;
+	}
+
+	private _supportsBlue: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Blue",
+	})
+	public get supportsBlue(): boolean {
+		return this._supportsBlue;
+	}
+
+	private _supportsAmber: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Amber",
+	})
+	public get supportsAmber(): boolean {
+		return this._supportsAmber;
+	}
+
+	private _supportsCyan: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Cyan",
+	})
+	public get supportsCyan(): boolean {
+		return this._supportsCyan;
+	}
+
+	private _supportsPurple: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Purple",
+	})
+	public get supportsPurple(): boolean {
+		return this._supportsPurple;
+	}
+
+	private _supportsIndex: boolean;
+	@ccValue()
+	@ccValueMetadata({
+		...ValueMetadata.ReadOnlyBoolean,
+		label: "Supports Indexed Color",
+	})
+	public get supportsIndex(): boolean {
+		return this._supportsIndex;
+	}
+
+	public toJSON(): JSONObject {
+		return super.toJSONInherited({
+			supportsWarmWhite: this._supportsWarmWhite,
+			supportsColdWhite: this._supportsColdWhite,
+			supportsRed: this._supportsRed,
+			supportsGreen: this._supportsGreen,
+			supportsBlue: this._supportsBlue,
+			supportsAmber: this._supportsAmber,
+			supportsCyan: this._supportsCyan,
+			supportsPurple: this._supportsPurple,
+			supportsIndex: this._supportsIndex,
+		});
+	}
+}
+
+@CCCommand(ColorSwitchCommand.SupportedGet)
+@expectedCCResponse(ColorSwitchCCSupportedReport)
+export class ColorSwitchCCSupportedGet extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions | CCCommandOptions,
+	) {
+		super(driver, options);
+	}
+}
+
+@CCCommand(ColorSwitchCommand.Report)
+export class ColorSwitchCCReport extends ColorSwitchCC {
+	public constructor(
+		driver: Driver,
+		options: CommandClassDeserializationOptions,
+	) {
+		super(driver, options);
+
+		// Docs say 'variable length', but the table shows 2 bytes.
+		validatePayload(this.payload.length >= 2);
+		this._colorComponent = this.payload[0];
+		this._value = this.payload[1];
+
+		// TODO: We probably want to store each color value
+		//	as a different ccValue.  How do we do that when
+		//	we get each color code separately in a single report?
+	}
+
+	private _colorComponent: ColorComponent;
+	public get colorComponent(): ColorComponent {
+		return this._colorComponent;
+	}
+
+	private _value: number;
+	public get value(): number {
+		return this._value;
+	}
+}
+
+interface ColorSwitchCCGetOptions extends CCCommandOptions {
+	colorComponent: ColorComponent;
+}
+
+@CCCommand(ColorSwitchCommand.Get)
+@expectedCCResponse(ColorSwitchCCReport)
```

Comment:
Ok, I forgot the one comment regarding the response check... You can basically copy the check from `BinarySensorCCGet`. You want to make sure that only `ColorSwitchCCReport` with a matching `colorComponent` are accepted as a response to the `Get`.

Reply:
Thanks, I got that in but I do not know the significance of isPositiveTransmitReport.  What does this represent?  Is the same check needed here?

```js
const testResponseForColorSwitchGet: CCResponsePredicate = (
	sent: ColorSwitchCCGet,
	received,
	isPositiveTransmitReport,
) => {
	// We expect a Binary Sensor Report that matches the requested sensor type (if a type was requested)
	return received instanceof ColorSwitchCCReport &&
		sent.colorComponent === received.colorComponent
		? "final"
		: // TODO: What is isPositiveTransmitReport?  Do I need it?
		isPositiveTransmitReport
		? "confirmation"
		: "unexpected";
};
```